lib/module.js

```diff
@@ -29,21 +27,24 @@ module.exports = function (moduleOptions) {
     registerValidator.call(this, options)
   }
 
-  this.nuxt.hook('build:extendRoutes', (routes) => {
-    processRoutes(routes)
+  this.nuxt.hook('modules:done', (moduleContainer) => {
```

Comment:
There is no need to register `module:done` hook, simply call `processRoutes` will be fine  
look at how nuxt-i18n done this

https://github.com/nuxt-community/nuxt-i18n/blob/e3e3f2d201b829f9b95d54b5e87d751ba48438a4/src/index.js#L62